Bluetooth: Mesh: DK Prov: Clarify UUID usage (#2992)
nrfconnect/sdk-nrf: include/bluetooth/mesh/dk_prov.rst

====

The provisioning handler uses :cpp:func:`hwinfo_get_device_id` from the ``hwinfo`` driver to generate a unique, deterministic UUID for each device.

> The UUID is used in the unprovisioned beacon as well as the broadcasted Mesh Provisioning Service data, and allows Provisioners to uniquely identify the device before starting provisioning.

Reviewer: ```suggestion
The UUID is used in the unprovisioned beacon and the broadcasted Mesh Provisioning Service data.
It allows the Provisioners to uniquely identify the device before starting provisioning.
```
